tests/helpers/test_mkfds.c

```diff
@@ -107,6 +107,14 @@ static void __attribute__((__noreturn__)) usage(FILE *out, int status)
 	exit(status);
 }
 
+#define XMEMDUP(PTR) xmemdup(PTR, sizeof(*(PTR)))
+static void *xmemdup(void *ptr, size_t size)
+{
+	void *r = xmalloc(size);
+	memcpy(r, ptr, size);
+	return r;
+}
+
```

Comment:
This is interesting. I can imagine this function in `include/xalloc.h` with __ul_calloc_size() attribute.  It would probably be more readable to not use the XMEMDUP() macro, but directly use xmemdup(ptr, sizeof(*ptr));

Reply:
Updated.